src/v/kafka/client/topic_cache.cc

```diff
@@ -57,7 +57,7 @@ topic_cache::leader(model::topic_partition tp) const {
         }
     }
     return ss::make_exception_future<model::node_id>(
-      partition_error(std::move(tp), error_code::unknown_topic_or_partition));
```

Comment:
I don't see the lifetime issue, this function has a copy and this is the last line.